src/main/java/com/google/sps/utility/GeoHash.java

```diff
@@ -0,0 +1,198 @@
+        bit = 0;
+        index = 0;
+      }
+    }
+
+    return geoHash;
+  }
+
+  /**
+   * Calculates the precision of the geohash given distance in Meters
+   * @param distance {int} Distance to determine precision
+   * @return {String} precision - The number of letters of the geohash
+   */
+  public static int determinePrecision(int distance) {
+    int precision = 0;
+    int initialDistance = COMMONS.MAX_GEOHASH_HEIGHT;
+    while(initialDistance >= distance && precision < COMMONS.MAX_GEOHASH_PRECISION) {
+      if (precision % 2 == 0) initialDistance /= 4;
+      else initialDistance /= 8;
+      precision++;
+    }
+    return precision;
+  }
+
+  /**
+   * Determines adjacent cell in given direction,
+   * @param geohash - Cell to which adjacent cell is required,
+   * @param direction - Direction from geohash (N/S/E/W).
+   * @return {String} GeoHash of adjacent cell,
+   */
+  private static String adjacentCell(String geohash, char direction) {
+    geohash = geohash.toLowerCase();
+    direction = Character.toLowerCase(direction);
+
+    if (geohash.length() == 0) return "";
+    if ("nsew".indexOf(direction) == -1) return "";
+
+    HashMap<Character, List<String>> neighbour = new HashMap<Character, List<String>>();
+    neighbour.put('n', Arrays.asList("p0r21436x8zb9dcf5h7kjnmqesgutwvy",
+            "bc01fg45238967deuvhjyznpkmstqrwx"));
+    neighbour.put('s', Arrays.asList("14365h7k9dcfesgujnmqp0r2twvyx8zb",
+            "238967debc01fg45kmstqrwxuvhjyznp"));
+    neighbour.put('e', Arrays.asList("bc01fg45238967deuvhjyznpkmstqrwx",
+            "p0r21436x8zb9dcf5h7kjnmqesgutwvy"));
+    neighbour.put('w', Arrays.asList("238967debc01fg45kmstqrwxuvhjyznp",
+            "14365h7k9dcfesgujnmqp0r2twvyx8zb"));
+
+    HashMap<Character, List<String>> border = new HashMap<Character, List<String>>();
+    border.put('n', Arrays.asList("prxz","bcfguvyz"));
+    border.put('s', Arrays.asList("028b","0145hjnp"));
+    border.put('e', Arrays.asList("bcfguvyz","prxz"));
+    border.put('w', Arrays.asList("0145hjnp","028b"));
+
+    // Last character of hash
+    char lastChar = geohash.charAt(geohash.length() - 1);
+    // Hash without last character
+    String parent = geohash.substring(0, geohash.length() - 1);
+
+    int type = geohash.length() % 2;
+
+    // Check for edge-cases which don't share common prefix
+    if (border.get(direction).get(type).indexOf(lastChar) != -1 && !parent.equals("")) {
+      parent = adjacentCell(parent, direction);
+    }
+
+    // append letter for direction to parent
+    return parent +
+            COMMONS.BASE32.charAt(neighbour
+                    .get(direction)
+                    .get(type)
+                    .indexOf(lastChar));
+  }
+
+  /**
+   * Returns all 8 adjacent cells to specified geohash.
+   * @param geohash {String} Geohash neighbours are required of.
+   * @return List of neighbours from a geohashed cell.
+   */
+  public static List<String> neighbourCells(String geohash) {
```

Comment:
```suggestion
  public static List<String> getNeighboringCells(String geohash) {
```
Maybe add "get" to the function name?